cmd/aries-js-worker/main.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"syscall/js"
+	"time"
+)
+
+// command is received from JS
+type command struct {
+	ID      string `json:"id"`
+	Pkg     string `json:"pkg"`
+	Fn      string `json:"fn"`
+	Payload string `json:"payload"`
+}
+
+// result is sent back to JS
+type result struct {
+	ID      string `json:"id"`
+	IsErr   bool   `json:"isErr"`
+	ErrMsg  string `json:"errMsg"`
+	Payload string `json:"payload"`
+}
+
+// All supported command handlers.
+func handlers() map[string]map[string]func(*command) *result {
+	return map[string]map[string]func(*command) *result{
+		"test": {
+			"echo":       echo,
+			"throwError": throwError,
+			"timeout":    timeout,
+		},
+	}
+}
+
+// main registers the 'handleMsg' function in the JS context's global scope to receive commands.
+// results are posted back to the 'handleResult' JS function.
+func main() {
+	input := make(chan *command)
+	output := make(chan *result)
+
+	go pipe(input, output)
+
+	go sendTo(output)
+
+	js.Global().Set("handleMsg", js.FuncOf(takeFrom(input)))
+
+	select {}
```

Comment:
We may need to consider a stop signal in a future PR.

Reply:
Yes, we may.